src/sidebar/BrowseView.js

```diff
@@ -80,6 +96,17 @@ class BrowseView extends Gtk.ScrolledWindow {
     }
   }
 
+  #onGestureClick = (gesture, n_press, x, y) => {
+    switch (gesture.get_current_button()) {
+      case Gdk.BUTTON_MIDDLE: {
+        const index = Math.floor((this._adj.value + y) / 38);
+        const uri = this._tree_model.get_row(index).item.uri;
+        this.activate_action("app.new-tab", new GLib.Variant("s", uri));
+        break;
+      }
+    }
+  };
+
```

Comment:
I would love to know if there's a better way to do this, but this easiest solution I could come up with. Maybe you could do something with ``compute_point`` but I couldn't figure anything out